src/MICore/Debugger.cs

```diff
@@ -316,8 +318,29 @@ protected virtual void OnStateChanged(string mode, string strresult)
             this.OnStateChanged(mode, _miResults.ParseResultList(strresult));
         }
 
+        private void SetGdbVersion()
+        {
+            if (string.IsNullOrWhiteSpace(_gdbVersion))
+            {
+                _gdbVersion = GdbVersionFromLog();
+            }
+        }
+
+        private void SetMajorGdbVersion()
+        {
+            int majorVersion = -1;
+            if (!string.IsNullOrWhiteSpace(_gdbVersion))
+            {
+                int.TryParse(_gdbVersion.Split('.').FirstOrDefault(), out majorVersion);
```

Comment:
>int [](http://example.com/codeflow?start=16&length=3)

Handle failures here?

Reply:
_gdbVersion.Split('.').FirstOrDefault() should either return a number, string or empty. If the TryParse fails, we would set the majorVersion to 0. I do not suspect an exception here, but I can wrap it inside a try catch block and log a warning.

Reply:
No reason have a try/catch, I was instead just suggesting that we should use the same major version if `_gdbVersion` is missing, then if TryParse returned false. So either initialize `majorVersion` to 0, or check the return value, and if it is false, assign it -1. Since the command factory defaults it to 0, I would probably suggest that.

Reply:
Done!